compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -273,7 +273,11 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
     val cls = mirroredType.classSymbol
     val useCompanion = cls.useCompanionAsSumMirror
 
-    if cls.isGenericSum(if useCompanion then cls.linkedClass else ctx.owner) then
+    def acceptable(tp: Type): Boolean = tp match
```

Comment:
I think this should be able to look through annotations/other decorations allowed on union types, e.g. the following still works (for `neg/scala_enum_testing.scala`):
```scala
summon[deriving.Mirror.SumOf[(Bar.A | Bar.B) @unchecked]]
```

Reply:
also I was able to bypass by doing a higher kinded mirror:
```scala
type Query[F[_]] = deriving.Mirror.Sum { type MirroredType[T] = F[T] }

scala> summon[Query[[X] =>> Bar.A | Bar.B]]
val res3: 
  scala.deriving.Mirror.Sum{
    MirroredMonoType = Bar.A | Bar.B; MirroredType[T] = Bar.A | Bar.B; 
      MirroredLabel = "Bar"
    ; MirroredElemTypes[T] = (Bar.A, Bar.B, Bar.C); 
      MirroredElemLabels = ("A", "B", "C")
  } = anon$1@4ecb9ab1
```

Reply:
I'll add a commit with a fix and tests for these